### PR TITLE
tensorOps changes

### PR DESCRIPTION
**Pull request by klevzoff**
Related to https://github.com/GEOSX/GEOSX/pull/1180

### CODE REVIEW

File: scripts/gdb-printers.py [hunk 1]
```diff
@@ -77,10 +77,21 @@ def data(self):
         return self.val['m_pointer']
 
 
+class MallocBufferPrinter(BufferPrinter):
```
Comment (corbett5) [inline scripts/gdb-printers.py:80]: I've never used these but it seems really cool. I don't imagine there's a good way to add a test, but would you mind at some point adding a short blurb to the sphinx? Like do I need to do anything to use it?

File: src/genericTensorOps.hpp [hunk 2]
```diff
@@ -539,6 +563,28 @@ void add( DST_MATRIX && LVARRAY_RESTRICT_REF dstMatrix,
...
+ * @brief Subtract @p value to @p dstVector.
+ * @tparam M The length of @p dstVector.
+ * @tparam DST_VECTOR The type of @p dstVector.
+ * @param dstVector The destination vector, of length M.
+ * @param value The value to subtract.
+ * @details Performs the operation @code dstVector[ i ] -= value @endcode
+ */
+template< std::ptrdiff_t M, typename DST_VECTOR >
+LVARRAY_HOST_DEVICE CONSTEXPR_WITHOUT_BOUNDS_CHECK inline
+void subtractScalar( DST_VECTOR && LVARRAY_RESTRICT_REF dstVector,
```
Comment (corbett5) [inline src/genericTensorOps.hpp:576]: I'm guessing you think it's worth having both `addScalar` and `subtractScalar`. Is there a reason not to have `subtractScalar` call `addScalar< M >( dstVector, -value )`?

File: src/typeManipulation.hpp [hunk 3]
```diff
@@ -79,6 +79,24 @@ public: \
...
 
+/**
+ * @brief Macro that expands to a static constexpr bool templated on a type that is only true when
+ *        the type has a nested type (or type alias) called @p NAME.
+ *        The name of the boolean variable is HasMemberType_ ## @p NAME.
+ * @param NAME The name of the type to look for.
+ */
+#define HAS_MEMBER_TYPE( NAME ) \
+  IS_VALID_EXPRESSION( HasMemberType_ ## NAME, CLASS, std::declval< typename CLASS::NAME >() )
+
```
Comment (corbett5) [inline src/typeManipulation.hpp:90]: 👍

**Review by corbett5 — APPROVED** [review]
Looks good! Would you mind adding tests for the tensorOps stuff. I could also add the tests for you, the tensorOps tests are kind of convoluted.

File: src/genericTensorOps.hpp (hunk 2 above)
Comment (klevzoff) [reply src/genericTensorOps.hpp:576]: No, it's just me being on autopilot :see_no_evil:

File: scripts/gdb-printers.py (hunk 1 above)
Comment (klevzoff) [reply scripts/gdb-printers.py:80]: Yes, I suppose a section under https://lvarray.readthedocs.io/en/latest/developmentAids.html would be appropriate

**Comment by klevzoff** [comment]
> Looks good! Would you mind adding tests for the tensorOps stuff. I could also add the tests for you, the tensorOps tests are kind of convoluted.

Sure! I will give it a try and ask you questions if I have any

File: scripts/gdb-printers.py (hunk 1 above)
Comment (corbett5) [reply scripts/gdb-printers.py:80]: Can you add it to the list of things on that page? Then once I get around to filling it out I might ask you to fill it in. Thanks.

Comment (klevzoff) [reply scripts/gdb-printers.py:80]: I ended up writing a few paragraphs about pretty printers, feel free to later use it however you see fit

File: unitTests/testTensorOpsOneSize.cpp [hunk 4]
```diff
@@ -433,6 +460,52 @@ class OneSizeTest : public ::testing::Test
         } );
   }
 
+  void testAiAj()
```
Comment (klevzoff) [inline unitTests/testTensorOpsOneSize.cpp:463]: I added tests for the two new `tensorOps` functions

**Pull request merged**